HW1/P4/P4.py

```diff
@@ -0,0 +1,40 @@
+import pyspark
+from P4_bfs import Graph_BFS
+
+sc = pyspark.SparkContext('local','Graph_BFS')
+
+lines=sc.textFile('source.csv')
+lines=lines.map(lambda l: l.split(','))
+
+# clean up, seperate the hero names and comic series
+lines_sp=lines.map(lambda l: (''.join(l[:-1]),l[-1]))
+tuple_heros=lines_sp
+tuple_comics=lines_sp.map(lambda l: (l[1], l[0]))
+
+# two rdds with hero as key and comics as key respectively
+hero_key_group=tuple_heros.groupByKey().mapValues(list)
+comics_key_group=tuple_comics.groupByKey().mapValues(list)
+
+# product them to get a larger table
+L_table=hero_key_group.cartesian(comics_key_group)
```

Comment:
A solution without cartesian may be faster. Why not join (hero, comic) pairs on 'comic'?